### PR TITLE
meta: update all databases & use oldest/latest naming scheme

### PR DESCRIPTION
**Pull request by ephys**
## Description of changes

This PR does the following:

- Update all images in `/dev` to match the ones we support in v7
- Use the 'oldest/latest' naming scheme for images in `/dev`
- Replace the `sushantdhiman/postgres` docker image with `postgis/postgis`, so we don't have to maintain it ourselves
- The postgres-latest image has been updated to postgres 12.
  - Waiting on https://github.com/brianc/node-postgres/issues/1927 to be resolved before moving to postgres 14
- Add `yarn test-integration-all`, `start-oldest`, `start-latest`, and `stop-all` scripts
  - Useful to `.only` a single test suite and run it on all dialects
- Merged `stop-x-latest` and `stop-x-oldest` because both commands are fast enough to reduce how many commands we provide
- Made the CI use `/dev` for postgres to reduce duplication. Let's see if this works
- ~~Removed `yarn test-unit-postgres-native` as unit tests don't interact with pg-native so shouldn't behave differently from `test-unit-postgres` (I may be wrong, I can revert this)~~

### CODE REVIEW

**Comment by WikiRik** [comment]
Thanks for this! I'm not so sure about the pg-native unit tests either. Might be better to revert that change for now until we've taken a better look at it

File: dev/mariadb/latest/start.sh
Comment (ephys) [inline dev/mariadb/latest/start.sh:16]: I removed this message because a similar one has been added to check-connection instead

**Comment by ephys** [comment]
I fail to see what I changed in db2 that would cause this issue. The only change there was supposed to be a folder rename. Very strange. Do you see something weird @WikiRik?

**Comment by ephys** [comment]
I downgraded postgres-latest to v12 until https://github.com/brianc/node-postgres/issues/1927 is fixed

File: .github/workflows/ci.yml
Comment (WikiRik) [inline .github/workflows/ci.yml:194]: Should we do this for the other dialects as well? It's a bit odd having this only for postgres

File: dev/mysql/oldest/docker-compose.yml
Comment (WikiRik) [inline dev/mysql/oldest/docker-compose.yml:4]: Should we specify the patch version here as well? Like we did with mysql-latest

File: package.json
Comment (WikiRik) [inline package.json:219]:
```suggestion
    "start-oldest": "concurrently \"npm:start-*-oldest\"",
```
Any reason not to do this?

File: package.json
Comment (WikiRik) [inline package.json:226]:
```suggestion
    "stop-all": "concurrently \"npm:stop-*(!all)\"",
```
Just like my previous suggestion

File: package.json
Comment (WikiRik) [inline package.json:249]: Do we want to add a `test-all-oldest` and `test-all-latest`?

File: package.json
Comment (WikiRik) [inline package.json:248]: Don't we want to make this `test-integration-oldest` and `test-integration-latest`?

File: test/integration/dialects/postgres/dao.test.js
Comment (WikiRik) [inline test/integration/dialects/postgres/dao.test.js:968]: Why this change?

File: package.json
Comment (ephys) [reply package.json:219]: DB2 requires root (with the current implementation at least), starting/stopping it in parallelp leads to the password prompt being lost in the noise

File: package.json
Comment (ephys) [reply package.json:249]: The databases use the same port, so the only difference is in which database version was started before running this command

File: test/integration/dialects/postgres/dao.test.js
Comment (ephys) [reply test/integration/dialects/postgres/dao.test.js:968]: Postgis was updated to V3 which includes more metadata, but V2 is still supported, which does not include it

File: dev/mysql/oldest/docker-compose.yml
Comment (ephys) [reply dev/mysql/oldest/docker-compose.yml:4]: I hesitated one which patch version to use:

- latest patch version v5.7.38
- or v5.7.0

Comment (WikiRik) [reply dev/mysql/oldest/docker-compose.yml:4]: Let's try out 5.7.0

**Comment by WikiRik** [comment]
About the pg-native unit tests; I took another look at them and I think we can safely remove those after all.

File: .github/workflows/ci.yml
Comment (ephys) [reply .github/workflows/ci.yml:194]: I started implementing these changes but I'd like to move it to its own PR that I can just discard if it doesn't work out

File: dev/mysql/oldest/docker-compose.yml
Comment (ephys) [reply dev/mysql/oldest/docker-compose.yml]: Done

**Comment by ephys** [comment]
> we can safely remove [native postgres unit tests] after all.

Done

File: package.json
Comment (ephys) [reply package.json:247]: Same reason as above, because oldest & latest run on the same port, they can't run at the same time, so `test-integration-oldest` & `test-integration-latest` would do the same thing as `test-integration-all`

**Comment by WikiRik** [comment]
What remains for this PR? And what are the next steps after merging this?

**Comment by ephys** [comment]
For this specific PR, I think everything is done.

For future PRs:
- Update postgres to v14 (blocked by https://github.com/brianc/node-postgres/issues/1927)
- Make Github Actions use the `/dev` folder instead of duplicating the database configs

**Review by WikiRik — DISMISSED** [review]

**Comment by ephys** [comment]
@WikiRik conflict resolution dismissed your review :(

**Review by WikiRik — APPROVED** [review]

**Pull request merged**
